sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -899,17 +900,17 @@ def _validate_input(self, X, y, incremental):
                          multi_output=True)
         if y.ndim == 2 and y.shape[1] == 1:
             y = column_or_1d(y, warn=True)
-        self.label_binarizer_.fit(y)
-
+            
         if not hasattr(self, 'classes_') or not incremental:
```

Comment:
When is this different to just `if not incremental`? (`classes_` are always set when called via `partial_fit`, no?)